File: LeetCode/Find_Pivot_Index.cpp
/* 
******
****** https://leetcode.com/problems/find-pivot-index/
******
*/

class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int sum = 0;
        int leftSum = 0;

        for (const int &n : nums)
        {
            sum += n;
        }

        for (int i = 0; i < nums.size(); i++)
        {
            if (sum - nums[i] - leftSum == leftSum)
                return i;
            leftSum += nums[i];
        }

        return -1;
    }
};

File: LeetCode/javascript/find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const pivotIndex = (nums) => {
  const sum = nums.reduce((prev, curr) => prev + curr, 0);
  let leftSum = 0;

  for (let i = 0; i < nums.length; i++) {
    if (sum - nums[i] - leftSum === leftSum) {
      return i;
    }
    leftSum += nums[i];
  }

  return -1;
};
